.github/workflows/nodejs.yml

```diff
@@ -27,3 +32,41 @@ jobs:
 
       - name: Run tests
         run: npm test
+
+  sonar:
+    name: Run eslint and sonar scanning
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: npm ci
```

Comment:
I think you should reuse the previously built artifact (built during "build" job), shouldn't you?
This artifact is tested and is about to be released - so this artifact should be analyzed, not a newly built one..

Reply:
I have now cached the `node_modules` folder, therefore there is no `npm ci` step needed for the `sonar` job